src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -141,3 +149,36 @@ def resolve_root_node_address(nodes: str) -> str:
         nodes = re.sub(r"\[(.*?)[,-].*\]", "\\1", nodes)  # Take the first node of every node range
         nodes = re.sub(r"\[(.*?)\]", "\\1", nodes)  # handle special case where node range is single number
         return nodes.split(" ")[0].split(",")[0]
+
+    @staticmethod
```

Comment:
I suggest making this a utility function in this file. Unless you have a specific reason to make it static.

Reply:
The reason is that this is not really a utility function that should be used anywhere else. It only makes sense in the context of this SLURM class and therefore is a staticmethod.

Reply:
This is mostly a stylistic choice, but so far we've been choosing module (aka utility) functions over static methods unless inheritance or a shared interface is desired. Not a strong suggestion though